components/kyma-environment-broker/internal/process/provisioning/event_hub_step.go

```diff
@@ -140,6 +147,41 @@ func (p *ProvisionAzureEventHubStep) Run(operation internal.ProvisioningOperatio
 	return operation, 0, nil
 }
 
+// getUniqueNSName will try to use suffix to generate a unique name for Event Hubs namespace, or generate a random one
+// if it already exists
+func (p *ProvisionAzureEventHubStep) getUniqueNSName(suffix string, azureClient azure.AzureInterface,
+	log logrus.FieldLogger) (string, error) {
+	uniqueName := fmt.Sprintf("skr-%s", suffix)
+	u, err := azureClient.CheckNamespaceAvailability(p.context, uniqueName)
+	if err != nil {
+		log.Errorf("Error while calling Azure Event Hubs client. Can't check namespace name availability. "+
+			"Error: %v", err)
+		return "", errors.Errorf("Error while calling Azure Event Hubs client. Can't check namespace name "+
+			"availability. Error: %v", err)
+	}
+
+	attempts := 10
+	for !u && attempts > 0 {
+		newName := fmt.Sprintf("skr-%s", uuid.New().String())
```

Comment:
in case the id skr-<suffix> ns is already used we dont use <suffix> anymore. Would it make sense to still use it in the new unique name to make it easier to map it to the cluster?
In case the newName would be to long you could easily use something other than UUID... i think we dont need the universal uniqueness at this point